### PR TITLE
Add Accept-CH-lifetime cache and CH header handling

### PR DESCRIPTION
**Pull request by yoavweiss**
This PR adds the concept of an Accept-CH cache as well as some handling for the `Accept-CH` and `Accept-CH-Lifetime` headers.
Tackles some of the concepts that need defining in https://github.com/whatwg/fetch/issues/726

***
<a href="https://whatpr.org/fetch/729.html" title="Last updated on May 22, 2018, 12:34 PM GMT (23d551f)">Preview</a> | <a href="https://whatpr.org/fetch/729/794dd54...23d551f.html" title="Last updated on May 22, 2018, 12:34 PM GMT (23d551f)">Diff</a>

### CODE REVIEW

File: fetch.bs [hunk 1]
```diff
@@ -1758,6 +1757,46 @@ of which is one of `<code>DPR</code>`, `<code>Save-Data</code>`, `<code>Viewport
...
+<p>An <dfn id=concept-accept-ch-cache>Accept-CH cache</dfn> is owned by the user agent and consists
+ of a collection of entries where each entry has these fields:
+
+<ul class=brief>
+ <li><dfn id=concept-accept-ch-cache-origin for=accept-ch-cache>origin</dfn> (an <a for=/>origin</a>)
+ <li><dfn id=concept-accept-ch-cache-max-age for=accept-ch-cache>max-age</dfn> (a number of seconds)
+ <li><dfn id=concept-accept-ch-cache-client-hints-list for=accept-ch-cache>client-hints list</dfn>
+</ul>
+
+<p>Entries must be removed after the seconds specified in the
```
Comment (igrigorik) [inline fetch.bs]: Do we need any extra clauses to ensure that this cache is purged when user clears site-data or similar?

File: fetch.bs [hunk 2]
```diff
@@ -2897,6 +2936,19 @@ with a <i>CORS flag</i> and <i>recursive flag</i>, run these steps:
  <li><p><a href=https://w3c.github.io/webappsec-csp/#set-response-csp-list>Set <var>internalResponse</var>'s CSP list</a>.
  [[!CSP]]
 
+ <li><p>If <var>response</var>'s
+ <a href=http://httpwg.org/http-extensions/client-hints.html#accept-ch>Accept-CH header</a>
+ is present, process the header and add its client-hints tokens to the <a>active document</a>'s
```
Comment (igrigorik) [inline fetch.bs]: Restrict to HTTPS only?

File: fetch.bs [hunk 3]
```diff
@@ -2897,6 +2936,19 @@ with a <i>CORS flag</i> and <i>recursive flag</i>, run these steps:
...
+ client-hints list.
+
+ <!--TODO: define the active document's client hints list -->
+
+ <li><p>If <var>response</var>'s
+ <a href=http://httpwg.org/http-extensions/client-hints.html#accept-ch-lifetime>Accept-CH-Lifetime header</a>
+ is present, call 
+ <a for=accept-ch-cache>create a new entry</a> with <var>response</var>'s <var>origin</var>, the
```
Comment (igrigorik) [inline fetch.bs]: Ditto, https-only?

File: fetch.bs (hunk 1 above)
Comment (yoavweiss) [reply fetch.bs]: Probably

File: fetch.bs (hunk 2 above)
Comment (yoavweiss) [reply fetch.bs]: Yeah

File: fetch.bs (hunk 1 above)
Comment (yoavweiss) [reply fetch.bs]: @mikewest what's the best way to hook this to clear-site-data or similar UI concepts (e.g. user cleared their cookies)? Looking at https://w3c.github.io/webappsec-clear-site-data/#clear-dom it seems to enumerate all the relevant things that need to be cleared? Should the ACHL cache do the same?

**Comment by yoavweiss** [comment]
Superseded by https://github.com/whatwg/html/pull/3774

**Pull request closed without merging**